app/controllers/account/pages_controller.rb

```diff
@@ -0,0 +1,19 @@
+class Account::PagesController < ApplicationController
+  def index
+    @pages = collection
+  end
+
+  def show
+    @page = resource
+  end
+
+  private
+
+  def resource
+    Page.find(params[:id])
```

Comment:
in future here code must looks something like

@page = @company.pages.find(params[:id])

for now it's ok